core/services/job/orm.go

```diff
@@ -186,27 +186,30 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 				}
 			}
 
+			if jb.OCROracleSpec.EVMChainID == nil {
+				// If unspecified, assume we're creating a job intended to run on default chain id
+				newChain, err := o.chainSet.Default()
+				if err != nil {
+					return err
+				}
+				if newChain == nil {
+					panic("Invariant violation:  chain set has no default chain id")
```

Comment:
Is it necessary to panic here? That seems severe when the method signature already has an error that could be leveraged.

Reply:
I guess we could.

My approach to invariant violations is that in general they should be panics, as a way of saving "if the code can get here, this shouldn't be released"... making it maximally unlikely we would release it with some kind of unpredictable data corruption.

I can see a case for making it an error though, as a way of not making a minor issue worse than it already is.  Do we have a strong precedent here in our codebase?

Reply:
The trouble is that considering this an invariant violation is pulling in a bunch of outside context unnecessarily, and requires upstream callers (test or prod) to be defensive about recovering from panics in order to produce user friendly errors.

Also, a case could be made that this is not in fact reporting an invariant violation - just upholding an invariant by refusing to violate it with the requested action. The existing state beforehand is _invalid_ and _misconfigured_ by our own definition - but the invariant wouldn't be violated unless things carried on with the nil.